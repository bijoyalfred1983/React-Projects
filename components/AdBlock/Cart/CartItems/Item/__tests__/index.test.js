import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Item } from '../index';

 

jest.mock(

  '../DeliveryOption',

  () =>

    function DeliveryOption() {

      return <span>Delivery Option</span>;

    }

);

jest.mock(

  '../CounterPart',

  () =>

    function CounterPart() {

      return <span>CounterPart</span>;

    }

);

 

describe('<CartItem />', () => {

  let PROPS = {};

const appData = {"staticLabelData":{"en_US":{"aaa":"aaa"}}};

  beforeEach(() => {

    PROPS = {

      appData,

      data: {

        storeItem: false,

        rawTotalPrice: 3698.73,

        estDeliveryTime: 'Est. Delivery: Feb  15-16',

        displayName: 'Duralast / Alternator',

        availability: 'In Stock',

        commerceItemId: 'ci343000004',

        techNotes: null,

        productNotes: '***900 Cold cranking amps (1000 cranking amps)***',

        vehicleFit: true,

        vehicleDescription: '2016 Chevrolet Silverado 1500',

        imageURL:

          'https://contentinfo.autozone.com/znetcs/product-info/en/US/dur/DL3655-6-9/image/1/',

        onSale: false,

        rebateAvailable: true,

        warrenty: '3 years',

        eligibleForNextDay: true,

        amount: 3698.73,

        '@class': 'com.autozone.diy.vo.AZCartCommerceItems',

        quantity: 27,

        salePrice: 0,

        dealAvailable: false,

        catalogRefId: '209336',

        application:

          'Down-stream sensor (located after catalytic converter). ESA Engine Code.',

        unitAmount: 136.99,

        location: 'front',

        partNumber: 'DL3655-6-9',

        quickNote: '425 Cold cranking Amps (530 cranking Amps)',

        currencyCode: 'USD',

        listPrice: 136.99,

        "skuPricingAndAvailability":{

            "lowestPrice":0,

            "corePriceAvailable":true,

            "rebateUrl":null,

            "vendorId":null,

            "bopusStore":false,

            "productLine":null,

            "storePickupCSSClass":null,

            "SRElegible":false,

            "configurableSkuMessage":null,

            "shipToHomeCSSClass":null,

            "storePickupLabel":"Store Pick Up",

            "shipToHomeLabel":"Ship To Home",

            "skuId":"319462",

            "highestPrice":0,

            "@class":"com.autozone.diy.commerce.pricing.AZPricingDealsInfo",

            "onlineStockLabel":null,

            "productId":null,

            "rebatesAvaialble":false,

            "shipToHomeStockLabel":"In Stock",

            "skuAddedTocart":false,

            "storePickupAvailable":false,

            "dealAvailable":false,

            "skuUnavailable":false,

            "storePickupStockLabel":"Out of Stock",

            "skuAvailabilityInfo":{

               "unAvailableInStoreAndVdpStore":false,

               "lowestPrice":null,

               "unavailOnlineQoh":false,

               "hubAvail":false,

               "storeAvail":true,

               "availableInOnline":false,

               "webCorePrice":null,

               "availableInVdpOnline":false,

               "shippingAvailabilityMessage":{

                  "eligibleForNextDay":false,

                  "@class":"com.autozone.diy.integration.shipping.ShippingAvailabilityMessage",

                  "inventoryMessage":null,

                  "holiday":false,

                  "cutOffTime":null,

                  "orderByMessage":null,

                  "calculatedDate":null,

                  "carrierHoliday":false,

                  "commerceItemId":null,

                  "qoHAvailable":false,

                  "displayDate":null,

                  "nextDayAvailableMessage":null,

                  "messageCutoffTime":null,

                  "messageDate":null

               },

               "availableInStore":true,

               "storeTotalPrice":"192.99",

               "orderByMessage":null,

               "showSearchAnotherStoreLink":false,

               "storeRetailPrice":"174.99",

               "unAvailableInOnlineAndVdpOnline":true,

               "unavailQoh":true,

               "eligibleForNextDay":false,

               "item":null,

               "@class":"com.autozone.diy.valueobject.SKUAvailabilityPriceInfo",

               "webRetailPrice":null,

               "webTotalPrice":null,

               "availableForDirectShipping":false,

               "exceptionItem":false,

               "storeCorePrice":"18.0",

               "availableInVdpStore":false,

               "messageCutOffTime":null,

               "nextDayAvailableMessage":null,

               "skuid":"319462"

            },

            "estimatedDeliveryMessage":"Not Available",

            "shipToHomeAvailable":false,

            "dealUrl":false,

            "configurableSku":false,

            "retailPrice":174.99,

            "categoryId":null,

            "corePrice":18

        }

      }

    };

  });

 

  test('renders correctly if all data is available', () => {

    PROPS.appData = {"staticLabelData":{"en_US":{

      "pagecontent_cart_corePopup_CoreDetails":"aaa",

      "label_cart_lineItem_Remove":"",

      "label_cart_lineItem_MailInRebateAvailable":"",

      "label_cart_lineItem_PartNo":"",

      "label_cart_lineItem_Location":"",

      "label_cart_lineItem_Warranty":"",

      "label_cart_lineItem_SeeProductNotes":"",

      "label_cart_lineItem_RebateAvailable":"",

      "label_cart_lineItem_Was":"",

      "label_cart_lineItem_Core":"",

      "label_cart_lineItem_WhatIsIt":"",

      "label_cart_lineItem_Fits":"",

      "label_cart_lineItem_SeeProductNotesSmall":"",

      "label_cart_notesOpen_Notes":"",

      "label_cart_notesOpen_Application":"",

      "label_cart_lineItem_Back":"",

      "label_cart_notesOpen_Notes":"",

      "label_cart_notesOpen_Application":""

    }}};

    const wrapper = mount(<Item {...PROPS} />);

    expect(toJson(wrapper)).toMatchSnapshot();

  });

 

  test('renders correctly if some data is unavailable', () => {

    PROPS.data = {

      ...PROPS.data,

      ...{

        productNotes: null,

        vehicleFit: false,

        vehicleDescription: null,

        rebateAvailable: false,

        warrenty: null,

        eligibleForNextDay: false,

        application: null

      }

    };

    const wrapper = mount(<Item {...PROPS} />);

    expect(toJson(wrapper)).toMatchSnapshot();

  });

 

  test('see product notes functionality', () => {

    const wrapper = mount(<Item {...PROPS} />);

    expect(wrapper.state('expanded')).toBe(false);

    wrapper

      .find('.seeProductNotes')

      .at(0)

      .props()

      .onClick();

    expect(wrapper.state('expanded')).toBe(true);

    wrapper

      .find('.seeProductNotes')

      .at(1)

      .props()

      .onClick();

    expect(wrapper.state('expanded')).toBe(false);

  });

 

  test('popup back button works', () => {

    const wrapper = mount(<Item {...PROPS} />);

    wrapper

      .find('.seeProductNotes')

      .at(0)

      .props()

      .onClick();

    expect(wrapper.state('expanded')).toBe(true);

    wrapper.update();

    wrapper

      .find('.backButton')

      .props()

      .onClick();

    expect(wrapper.state('expanded')).toBe(false);

  });

});