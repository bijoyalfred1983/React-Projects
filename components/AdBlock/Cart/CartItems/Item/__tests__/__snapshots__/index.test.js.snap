// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> renders correctly if all data is available 1`] = `
<Item
  data={
    Object {
      "@class": "com.autozone.diy.vo.AZCartCommerceItems",
      "amount": 3698.73,
      "application": "Down-stream sensor (located after catalytic converter). ESA Engine Code.",
      "availability": "In Stock",
      "catalogRefId": "209336",
      "commerceItemId": "ci343000004",
      "currencyCode": "USD",
      "dealAvailable": false,
      "displayName": "Duralast / Alternator",
      "eligibleForNextDay": true,
      "estDeliveryTime": "Est. Delivery: Feb  15-16",
      "imageURL": "https://contentinfo.autozone.com/znetcs/product-info/en/US/dur/DL3655-6-9/image/1/",
      "listPrice": 136.99,
      "location": "front",
      "onSale": false,
      "partNumber": "DL3655-6-9",
      "productNotes": "***900 Cold cranking amps (1000 cranking amps)***",
      "quantity": 27,
      "quickNote": "425 Cold cranking Amps (530 cranking Amps)",
      "rawTotalPrice": 3698.73,
      "rebateAvailable": true,
      "salePrice": 0,
      "skuPricingAndAvailability": Object {
        "@class": "com.autozone.diy.commerce.pricing.AZPricingDealsInfo",
        "SRElegible": false,
        "bopusStore": false,
        "categoryId": null,
        "configurableSku": false,
        "configurableSkuMessage": null,
        "corePrice": 18,
        "corePriceAvailable": true,
        "dealAvailable": false,
        "dealUrl": false,
        "estimatedDeliveryMessage": "Not Available",
        "highestPrice": 0,
        "lowestPrice": 0,
        "onlineStockLabel": null,
        "productId": null,
        "productLine": null,
        "rebateUrl": null,
        "rebatesAvaialble": false,
        "retailPrice": 174.99,
        "shipToHomeAvailable": false,
        "shipToHomeCSSClass": null,
        "shipToHomeLabel": "Ship To Home",
        "shipToHomeStockLabel": "In Stock",
        "skuAddedTocart": false,
        "skuAvailabilityInfo": Object {
          "@class": "com.autozone.diy.valueobject.SKUAvailabilityPriceInfo",
          "availableForDirectShipping": false,
          "availableInOnline": false,
          "availableInStore": true,
          "availableInVdpOnline": false,
          "availableInVdpStore": false,
          "eligibleForNextDay": false,
          "exceptionItem": false,
          "hubAvail": false,
          "item": null,
          "lowestPrice": null,
          "messageCutOffTime": null,
          "nextDayAvailableMessage": null,
          "orderByMessage": null,
          "shippingAvailabilityMessage": Object {
            "@class": "com.autozone.diy.integration.shipping.ShippingAvailabilityMessage",
            "calculatedDate": null,
            "carrierHoliday": false,
            "commerceItemId": null,
            "cutOffTime": null,
            "displayDate": null,
            "eligibleForNextDay": false,
            "holiday": false,
            "inventoryMessage": null,
            "messageCutoffTime": null,
            "messageDate": null,
            "nextDayAvailableMessage": null,
            "orderByMessage": null,
            "qoHAvailable": false,
          },
          "showSearchAnotherStoreLink": false,
          "skuid": "319462",
          "storeAvail": true,
          "storeCorePrice": "18.0",
          "storeRetailPrice": "174.99",
          "storeTotalPrice": "192.99",
          "unAvailableInOnlineAndVdpOnline": true,
          "unAvailableInStoreAndVdpStore": false,
          "unavailOnlineQoh": false,
          "unavailQoh": true,
          "webCorePrice": null,
          "webRetailPrice": null,
          "webTotalPrice": null,
        },
        "skuId": "319462",
        "skuUnavailable": false,
        "storePickupAvailable": false,
        "storePickupCSSClass": null,
        "storePickupLabel": "Store Pick Up",
        "storePickupStockLabel": "Out of Stock",
        "vendorId": null,
      },
      "storeItem": false,
      "techNotes": null,
      "unitAmount": 136.99,
      "vehicleDescription": "2016 Chevrolet Silverado 1500",
      "vehicleFit": true,
      "warrenty": "3 years",
    }
  }
>
  <div
    className="cardBlock"
  >
    <span
      className="remove desktop"
    >
      remove
    </span>
    <div
      className="aboveHeader mobile"
    >
      <img
        alt="next day delivery"
        className="nextDayDelivery"
        src="/images/next-day-delivery.png"
      />
      <div
        className="rebate"
      >
        <img
          alt="rebate"
          src="/images/info.png"
          style={
            Object {
              "marginRight": "4px",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "verticalAlign": "middle",
            }
          }
        >
          Mail-In Rebate Available
        </span>
      </div>
    </div>
    <div
      className="header"
    >
      <div>
        <span
          className="itemName"
        >
          Duralast / Alternator
        </span>
        <img
          alt="next day delivery"
          className="nextDayDelivery desktop"
          src="/images/next-day-delivery.png"
        />
      </div>
      <span
        className="quickNote"
      >
        425 Cold cranking Amps (530 cranking Amps)
      </span>
    </div>
    <div
      className="row content itemInfo"
    >
      <div
        className="col l5 m4 s12"
      >
        <div
          className="col l4 m5 s5"
        >
          <img
            alt="thumbnail"
            className="thumbnail"
            src="https://contentinfo.autozone.com/znetcs/product-info/en/US/dur/DL3655-6-9/image/1/"
          />
        </div>
        <div
          className="col l8 m7 s7"
        >
          <div>
            Part #
             
            <span
              className="itemInfoValue"
            >
              DL3655-6-9
            </span>
          </div>
          <div>
            Location
             
            <span
              className="itemInfoValue"
            >
              front
            </span>
          </div>
          <div>
            Warranty
             
            <span
              className="itemInfoValue warranty"
            >
              3 years
            </span>
          </div>
          <div
            className="seeProductNotes mobile"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            See Product Notes
          </div>
          <div
            className="rebate desktop"
          >
            Rebate Available
          </div>
        </div>
      </div>
      <div
        className="col l3 m3 s12"
      >
        <DeliveryOption
          data={
            Object {
              "@class": "com.autozone.diy.vo.AZCartCommerceItems",
              "amount": 3698.73,
              "application": "Down-stream sensor (located after catalytic converter). ESA Engine Code.",
              "availability": "In Stock",
              "catalogRefId": "209336",
              "commerceItemId": "ci343000004",
              "currencyCode": "USD",
              "dealAvailable": false,
              "displayName": "Duralast / Alternator",
              "eligibleForNextDay": true,
              "estDeliveryTime": "Est. Delivery: Feb  15-16",
              "imageURL": "https://contentinfo.autozone.com/znetcs/product-info/en/US/dur/DL3655-6-9/image/1/",
              "listPrice": 136.99,
              "location": "front",
              "onSale": false,
              "partNumber": "DL3655-6-9",
              "productNotes": "***900 Cold cranking amps (1000 cranking amps)***",
              "quantity": 27,
              "quickNote": "425 Cold cranking Amps (530 cranking Amps)",
              "rawTotalPrice": 3698.73,
              "rebateAvailable": true,
              "salePrice": 0,
              "skuPricingAndAvailability": Object {
                "@class": "com.autozone.diy.commerce.pricing.AZPricingDealsInfo",
                "SRElegible": false,
                "bopusStore": false,
                "categoryId": null,
                "configurableSku": false,
                "configurableSkuMessage": null,
                "corePrice": 18,
                "corePriceAvailable": true,
                "dealAvailable": false,
                "dealUrl": false,
                "estimatedDeliveryMessage": "Not Available",
                "highestPrice": 0,
                "lowestPrice": 0,
                "onlineStockLabel": null,
                "productId": null,
                "productLine": null,
                "rebateUrl": null,
                "rebatesAvaialble": false,
                "retailPrice": 174.99,
                "shipToHomeAvailable": false,
                "shipToHomeCSSClass": null,
                "shipToHomeLabel": "Ship To Home",
                "shipToHomeStockLabel": "In Stock",
                "skuAddedTocart": false,
                "skuAvailabilityInfo": Object {
                  "@class": "com.autozone.diy.valueobject.SKUAvailabilityPriceInfo",
                  "availableForDirectShipping": false,
                  "availableInOnline": false,
                  "availableInStore": true,
                  "availableInVdpOnline": false,
                  "availableInVdpStore": false,
                  "eligibleForNextDay": false,
                  "exceptionItem": false,
                  "hubAvail": false,
                  "item": null,
                  "lowestPrice": null,
                  "messageCutOffTime": null,
                  "nextDayAvailableMessage": null,
                  "orderByMessage": null,
                  "shippingAvailabilityMessage": Object {
                    "@class": "com.autozone.diy.integration.shipping.ShippingAvailabilityMessage",
                    "calculatedDate": null,
                    "carrierHoliday": false,
                    "commerceItemId": null,
                    "cutOffTime": null,
                    "displayDate": null,
                    "eligibleForNextDay": false,
                    "holiday": false,
                    "inventoryMessage": null,
                    "messageCutoffTime": null,
                    "messageDate": null,
                    "nextDayAvailableMessage": null,
                    "orderByMessage": null,
                    "qoHAvailable": false,
                  },
                  "showSearchAnotherStoreLink": false,
                  "skuid": "319462",
                  "storeAvail": true,
                  "storeCorePrice": "18.0",
                  "storeRetailPrice": "174.99",
                  "storeTotalPrice": "192.99",
                  "unAvailableInOnlineAndVdpOnline": true,
                  "unAvailableInStoreAndVdpStore": false,
                  "unavailOnlineQoh": false,
                  "unavailQoh": true,
                  "webCorePrice": null,
                  "webRetailPrice": null,
                  "webTotalPrice": null,
                },
                "skuId": "319462",
                "skuUnavailable": false,
                "storePickupAvailable": false,
                "storePickupCSSClass": null,
                "storePickupLabel": "Store Pick Up",
                "storePickupStockLabel": "Out of Stock",
                "vendorId": null,
              },
              "storeItem": false,
              "techNotes": null,
              "unitAmount": 136.99,
              "vehicleDescription": "2016 Chevrolet Silverado 1500",
              "vehicleFit": true,
              "warrenty": "3 years",
            }
          }
        >
          <div
            className="deliveryOption"
          >
            <form
              className="row display_flex"
            >
              <div
                className="pickUpStore grey_inactive"
              >
                <div
                  className="display_flex"
                >
                  <input
                    checked={false}
                    className="test5 pickup
                notAllowed"
                    disabled={true}
                    id="radio-1"
                    name="myRadio"
                    onChange={[Function]}
                    type="radio"
                    value="ship"
                  />
                </div>
                <div
                  className="storeLeft"
                >
                  <div
                    className="store_pickUp"
                    id="test5"
                  >
                    STORE-PICK-UP
                  </div>
                  <div
                    className="redFont"
                    id="test5"
                  >
                    Out of Stock
                  </div>
                </div>
              </div>
              <div
                className="shipHomeStore white_active"
              >
                <div>
                  <input
                    checked={true}
                    className="with-gap test51 pickup
                  notAllowed"
                    disabled={true}
                    id="test6"
                    name="myRadio"
                    onChange={[Function]}
                    type="radio"
                    value="store"
                  />
                </div>
                <div
                  className="storeLeft1"
                >
                  <div
                    className="store_pickUp"
                  >
                    SHIP-TO-HOME
                  </div>
                  <div
                    className="greenFont"
                  >
                    In Stock
                  </div>
                </div>
              </div>
            </form>
          </div>
        </DeliveryOption>
      </div>
      <div
        className="col l2 m3 s6"
      />
      <div
        className="col l2 m2 s6"
      >
        <div
          className="right-align"
        >
          <div>
            <div
              className="was"
            >
              WAS
              <span
                className="strike"
              >
                $
                174.99
              </span>
            </div>
            <div
              style={
                Object {
                  "marginTop": "5px",
                }
              }
            >
              <span
                className="priceSub"
              >
                $
              </span>
              <span
                className="price"
              >
                136
              </span>
              <span
                className="priceSub"
              >
                99
              </span>
            </div>
          </div>
          <div
            className="corePrice"
          >
            <span>
              <span>
                + 
              </span>
              Core $
              18
            </span>
            <Popup
              hoverable={true}
              on="hover"
              position="top left"
              trigger={
                <img
                  alt="info"
                  src="/images/info.png"
                  style={
                    Object {
                      "marginLeft": "5px",
                    }
                  }
                />
              }
            >
              <Portal
                closeOnDocumentClick={true}
                closeOnEscape={true}
                closeOnPortalMouseLeave={true}
                closeOnTriggerMouseLeave={true}
                eventPool="default"
                mouseEnterDelay={50}
                mouseLeaveDelay={70}
                onClose={[Function]}
                onMount={[Function]}
                onOpen={[Function]}
                onUnmount={[Function]}
                openOnTriggerClick={true}
                openOnTriggerMouseEnter={true}
                trigger={
                  <img
                    alt="info"
                    src="/images/info.png"
                    style={
                      Object {
                        "marginLeft": "5px",
                      }
                    }
                  />
                }
              >
                <Ref
                  innerRef={[Function]}
                >
                  <img
                    alt="info"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    src="/images/info.png"
                    style={
                      Object {
                        "marginLeft": "5px",
                      }
                    }
                  />
                </Ref>
              </Portal>
            </Popup>
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer"
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <span
          className="vehicle"
        >
          <span
            style={
              Object {
                "display": "table",
              }
            }
          >
            <img
              alt="fits"
              height="14"
              src="/images/fits.png"
              width="14"
            />
            <span
              className="fits"
            >
              FITS
            </span>
          </span>
          <span
            className="vehicleDescription"
          >
            2016 Chevrolet Silverado 1500
          </span>
        </span>
        <span
          className="seeProductNotes desktop"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex={0}
        >
          see product notes
          <img
            alt="expand"
            height="18px"
            src="/images/down-arrow.png"
            style={
              Object {
                "float": "right",
                "marginLeft": "10px",
              }
            }
            width="18px"
          />
        </span>
      </div>
    </div>
     
  </div>
</Item>
`;

exports[`<CartItem /> renders correctly if some data is unavailable 1`] = `
<Item
  data={
    Object {
      "@class": "com.autozone.diy.vo.AZCartCommerceItems",
      "amount": 3698.73,
      "application": null,
      "availability": "In Stock",
      "catalogRefId": "209336",
      "commerceItemId": "ci343000004",
      "currencyCode": "USD",
      "dealAvailable": false,
      "displayName": "Duralast / Alternator",
      "eligibleForNextDay": false,
      "estDeliveryTime": "Est. Delivery: Feb  15-16",
      "imageURL": "https://contentinfo.autozone.com/znetcs/product-info/en/US/dur/DL3655-6-9/image/1/",
      "listPrice": 136.99,
      "location": "front",
      "onSale": false,
      "partNumber": "DL3655-6-9",
      "productNotes": null,
      "quantity": 27,
      "quickNote": "425 Cold cranking Amps (530 cranking Amps)",
      "rawTotalPrice": 3698.73,
      "rebateAvailable": false,
      "salePrice": 0,
      "skuPricingAndAvailability": Object {
        "@class": "com.autozone.diy.commerce.pricing.AZPricingDealsInfo",
        "SRElegible": false,
        "bopusStore": false,
        "categoryId": null,
        "configurableSku": false,
        "configurableSkuMessage": null,
        "corePrice": 18,
        "corePriceAvailable": true,
        "dealAvailable": false,
        "dealUrl": false,
        "estimatedDeliveryMessage": "Not Available",
        "highestPrice": 0,
        "lowestPrice": 0,
        "onlineStockLabel": null,
        "productId": null,
        "productLine": null,
        "rebateUrl": null,
        "rebatesAvaialble": false,
        "retailPrice": 174.99,
        "shipToHomeAvailable": false,
        "shipToHomeCSSClass": null,
        "shipToHomeLabel": "Ship To Home",
        "shipToHomeStockLabel": "In Stock",
        "skuAddedTocart": false,
        "skuAvailabilityInfo": Object {
          "@class": "com.autozone.diy.valueobject.SKUAvailabilityPriceInfo",
          "availableForDirectShipping": false,
          "availableInOnline": false,
          "availableInStore": true,
          "availableInVdpOnline": false,
          "availableInVdpStore": false,
          "eligibleForNextDay": false,
          "exceptionItem": false,
          "hubAvail": false,
          "item": null,
          "lowestPrice": null,
          "messageCutOffTime": null,
          "nextDayAvailableMessage": null,
          "orderByMessage": null,
          "shippingAvailabilityMessage": Object {
            "@class": "com.autozone.diy.integration.shipping.ShippingAvailabilityMessage",
            "calculatedDate": null,
            "carrierHoliday": false,
            "commerceItemId": null,
            "cutOffTime": null,
            "displayDate": null,
            "eligibleForNextDay": false,
            "holiday": false,
            "inventoryMessage": null,
            "messageCutoffTime": null,
            "messageDate": null,
            "nextDayAvailableMessage": null,
            "orderByMessage": null,
            "qoHAvailable": false,
          },
          "showSearchAnotherStoreLink": false,
          "skuid": "319462",
          "storeAvail": true,
          "storeCorePrice": "18.0",
          "storeRetailPrice": "174.99",
          "storeTotalPrice": "192.99",
          "unAvailableInOnlineAndVdpOnline": true,
          "unAvailableInStoreAndVdpStore": false,
          "unavailOnlineQoh": false,
          "unavailQoh": true,
          "webCorePrice": null,
          "webRetailPrice": null,
          "webTotalPrice": null,
        },
        "skuId": "319462",
        "skuUnavailable": false,
        "storePickupAvailable": false,
        "storePickupCSSClass": null,
        "storePickupLabel": "Store Pick Up",
        "storePickupStockLabel": "Out of Stock",
        "vendorId": null,
      },
      "storeItem": false,
      "techNotes": null,
      "unitAmount": 136.99,
      "vehicleDescription": null,
      "vehicleFit": false,
      "warrenty": null,
    }
  }
>
  <div
    className="cardBlock"
  >
    <span
      className="remove desktop"
    >
      remove
    </span>
    <div
      className="header"
    >
      <div>
        <span
          className="itemName"
        >
          Duralast / Alternator
        </span>
      </div>
      <span
        className="quickNote"
      >
        425 Cold cranking Amps (530 cranking Amps)
      </span>
    </div>
    <div
      className="row content itemInfo"
    >
      <div
        className="col l5 m4 s12"
      >
        <div
          className="col l4 m5 s5"
        >
          <img
            alt="thumbnail"
            className="thumbnail"
            src="https://contentinfo.autozone.com/znetcs/product-info/en/US/dur/DL3655-6-9/image/1/"
          />
        </div>
        <div
          className="col l8 m7 s7"
        >
          <div>
            Part #
             
            <span
              className="itemInfoValue"
            >
              DL3655-6-9
            </span>
          </div>
          <div>
            Location
             
            <span
              className="itemInfoValue"
            >
              front
            </span>
          </div>
          <div>
            Warranty
             
            <span
              className="itemInfoValue warranty"
            />
          </div>
        </div>
      </div>
      <div
        className="col l3 m3 s12"
      >
        <DeliveryOption
          data={
            Object {
              "@class": "com.autozone.diy.vo.AZCartCommerceItems",
              "amount": 3698.73,
              "application": null,
              "availability": "In Stock",
              "catalogRefId": "209336",
              "commerceItemId": "ci343000004",
              "currencyCode": "USD",
              "dealAvailable": false,
              "displayName": "Duralast / Alternator",
              "eligibleForNextDay": false,
              "estDeliveryTime": "Est. Delivery: Feb  15-16",
              "imageURL": "https://contentinfo.autozone.com/znetcs/product-info/en/US/dur/DL3655-6-9/image/1/",
              "listPrice": 136.99,
              "location": "front",
              "onSale": false,
              "partNumber": "DL3655-6-9",
              "productNotes": null,
              "quantity": 27,
              "quickNote": "425 Cold cranking Amps (530 cranking Amps)",
              "rawTotalPrice": 3698.73,
              "rebateAvailable": false,
              "salePrice": 0,
              "skuPricingAndAvailability": Object {
                "@class": "com.autozone.diy.commerce.pricing.AZPricingDealsInfo",
                "SRElegible": false,
                "bopusStore": false,
                "categoryId": null,
                "configurableSku": false,
                "configurableSkuMessage": null,
                "corePrice": 18,
                "corePriceAvailable": true,
                "dealAvailable": false,
                "dealUrl": false,
                "estimatedDeliveryMessage": "Not Available",
                "highestPrice": 0,
                "lowestPrice": 0,
                "onlineStockLabel": null,
                "productId": null,
                "productLine": null,
                "rebateUrl": null,
                "rebatesAvaialble": false,
                "retailPrice": 174.99,
                "shipToHomeAvailable": false,
                "shipToHomeCSSClass": null,
                "shipToHomeLabel": "Ship To Home",
                "shipToHomeStockLabel": "In Stock",
                "skuAddedTocart": false,
                "skuAvailabilityInfo": Object {
                  "@class": "com.autozone.diy.valueobject.SKUAvailabilityPriceInfo",
                  "availableForDirectShipping": false,
                  "availableInOnline": false,
                  "availableInStore": true,
                  "availableInVdpOnline": false,
                  "availableInVdpStore": false,
                  "eligibleForNextDay": false,
                  "exceptionItem": false,
                  "hubAvail": false,
                  "item": null,
                  "lowestPrice": null,
                  "messageCutOffTime": null,
                  "nextDayAvailableMessage": null,
                  "orderByMessage": null,
                  "shippingAvailabilityMessage": Object {
                    "@class": "com.autozone.diy.integration.shipping.ShippingAvailabilityMessage",
                    "calculatedDate": null,
                    "carrierHoliday": false,
                    "commerceItemId": null,
                    "cutOffTime": null,
                    "displayDate": null,
                    "eligibleForNextDay": false,
                    "holiday": false,
                    "inventoryMessage": null,
                    "messageCutoffTime": null,
                    "messageDate": null,
                    "nextDayAvailableMessage": null,
                    "orderByMessage": null,
                    "qoHAvailable": false,
                  },
                  "showSearchAnotherStoreLink": false,
                  "skuid": "319462",
                  "storeAvail": true,
                  "storeCorePrice": "18.0",
                  "storeRetailPrice": "174.99",
                  "storeTotalPrice": "192.99",
                  "unAvailableInOnlineAndVdpOnline": true,
                  "unAvailableInStoreAndVdpStore": false,
                  "unavailOnlineQoh": false,
                  "unavailQoh": true,
                  "webCorePrice": null,
                  "webRetailPrice": null,
                  "webTotalPrice": null,
                },
                "skuId": "319462",
                "skuUnavailable": false,
                "storePickupAvailable": false,
                "storePickupCSSClass": null,
                "storePickupLabel": "Store Pick Up",
                "storePickupStockLabel": "Out of Stock",
                "vendorId": null,
              },
              "storeItem": false,
              "techNotes": null,
              "unitAmount": 136.99,
              "vehicleDescription": null,
              "vehicleFit": false,
              "warrenty": null,
            }
          }
        >
          <div
            className="deliveryOption"
          >
            <form
              className="row display_flex"
            >
              <div
                className="pickUpStore grey_inactive"
              >
                <div
                  className="display_flex"
                >
                  <input
                    checked={false}
                    className="test5 pickup
                notAllowed"
                    disabled={true}
                    id="radio-1"
                    name="myRadio"
                    onChange={[Function]}
                    type="radio"
                    value="ship"
                  />
                </div>
                <div
                  className="storeLeft"
                >
                  <div
                    className="store_pickUp"
                    id="test5"
                  >
                    STORE-PICK-UP
                  </div>
                  <div
                    className="redFont"
                    id="test5"
                  >
                    Out of Stock
                  </div>
                </div>
              </div>
              <div
                className="shipHomeStore white_active"
              >
                <div>
                  <input
                    checked={true}
                    className="with-gap test51 pickup
                  notAllowed"
                    disabled={true}
                    id="test6"
                    name="myRadio"
                    onChange={[Function]}
                    type="radio"
                    value="store"
                  />
                </div>
                <div
                  className="storeLeft1"
                >
                  <div
                    className="store_pickUp"
                  >
                    SHIP-TO-HOME
                  </div>
                  <div
                    className="greenFont"
                  >
                    In Stock
                  </div>
                </div>
              </div>
            </form>
          </div>
        </DeliveryOption>
      </div>
      <div
        className="col l2 m3 s6"
      />
      <div
        className="col l2 m2 s6"
      >
        <div
          className="right-align"
        >
          <div>
            <div
              className="was"
            >
              WAS
              <span
                className="strike"
              >
                $
                174.99
              </span>
            </div>
            <div
              style={
                Object {
                  "marginTop": "5px",
                }
              }
            >
              <span
                className="priceSub"
              >
                $
              </span>
              <span
                className="price"
              >
                136
              </span>
              <span
                className="priceSub"
              >
                99
              </span>
            </div>
          </div>
          <div
            className="corePrice"
          >
            <span>
              <span>
                + 
              </span>
              Core $
              18
            </span>
            <Popup
              hoverable={true}
              on="hover"
              position="top left"
              trigger={
                <img
                  alt="info"
                  src="/images/info.png"
                  style={
                    Object {
                      "marginLeft": "5px",
                    }
                  }
                />
              }
            >
              <Portal
                closeOnDocumentClick={true}
                closeOnEscape={true}
                closeOnPortalMouseLeave={true}
                closeOnTriggerMouseLeave={true}
                eventPool="default"
                mouseEnterDelay={50}
                mouseLeaveDelay={70}
                onClose={[Function]}
                onMount={[Function]}
                onOpen={[Function]}
                onUnmount={[Function]}
                openOnTriggerClick={true}
                openOnTriggerMouseEnter={true}
                trigger={
                  <img
                    alt="info"
                    src="/images/info.png"
                    style={
                      Object {
                        "marginLeft": "5px",
                      }
                    }
                  />
                }
              >
                <Ref
                  innerRef={[Function]}
                >
                  <img
                    alt="info"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    src="/images/info.png"
                    style={
                      Object {
                        "marginLeft": "5px",
                      }
                    }
                  />
                </Ref>
              </Portal>
            </Popup>
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer"
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <span
          className="vehicle"
        >
          <span
            className="vehicleDescription"
          >
            No vehicle assigned
          </span>
        </span>
      </div>
    </div>
     
  </div>
</Item>
`;
